Quandl.Excel.UDF.Functions/Helpers/SheetHelper.cs

```diff
@@ -68,23 +125,21 @@ public void PopulateData(Range currentFormulaCell)
             {
                 Trace.WriteLine(e.Message);
 
-                // Release Mutex to allow another function to write data.
-                DataWriteMutex.ReleaseMutex();
 
                 // The excel RPC server is busy. We need to wait and then retry (RPC_E_SERVERCALL_RETRYLATER or VBA_E_IGNORE)
                 if (e.HResult == Exception.RPC_E_SERVERCALL_RETRYLATER || e.HResult == Exception.VBA_E_IGNORE)
                 {
                     Thread.Sleep(RetryWaitTimeMs);
-                    PopulateData(currentFormulaCell);
+                    Populate();
                 }
 
                 throw;
             }
             finally
             {
                 // Release Mutex to allow another function to write data.
-                DataWriteMutex.ReleaseMutex();
-            }
+                //DataWriteMutex.ReleaseMutex();
+            }*/
```

Comment:
could you clean the code which is comments out? and for all other part which is comments out, I think we can easily trace them by git

Reply:
UDF were not migrated, rather I corrected resource usage. This specific mutex however should not have been removed; I removed it because I was hoping to execute both workers on foreground thread explicitly, however that did not work out and QTABLE still executes on background thread. A better solution though is to move the mutex to `ExcelExecutionHelper.ExecuteWithAutoRetry` function.

Reply:
I decided to only lock executions from background thread. Mutex on the foreground thread is ugly.